jwst/flatfield/flat_field.py

```diff
@@ -1119,9 +1174,28 @@ def combine_dq(f_flat_dq, s_flat_dq, d_flat_dq, default_shape):
         flat_dq = dq_list[0].copy()
     elif n_dq == 2:
         flat_dq = np.bitwise_or(dq_list[0], dq_list[1])
+        bad1 = np.bitwise_and(dq_list[0], dqflags.pixel['DO_NOT_USE'])
+        bad2 = np.bitwise_and(dq_list[1], dqflags.pixel['DO_NOT_USE'])
+        iflag = np.where( (bad1==1) & (bad2 ==1))
+
+        # locate places that flat_dq is set to DO_NOT_USE and reset to UNRELIABLE FLAT
+        iloc = np.where(np.bitwise_and(flat_dq, dqflags.pixel['DO_NOT_USE']))
+        flat_dq[iloc] = dqflags.pixel['UNRELIABLE_FLAT']
+        # now only set DO_NOT_USE to pixels that are set in both flats as DO_NOT_USE
+        flat_dq[iflag] = np.bitwise_or(flat_dq[iflag],dqflags.pixel['DO_NOT_USE'])
     elif n_dq == 3:
```

Comment:
Isn't there some more elegant way to do all of this checking of dq values from any of 1, 2, or 3 flats, like just a single loop that has a range of n_dq and combines values from all the individual dq arrays that exist? As opposed to having hardwired code blocks to handle each instance of 0, 1, 2, or 3 dq arrays.